### PR TITLE
JEL-340: Rework message tooling using a specific bus

### PR DESCRIPTION
**Pull request by LaurentPetard**
EE: https://github.com/akeneo/pim-enterprise-dev/pull/17651

### CODE REVIEW

File: src/Akeneo/Tool/Bundle/MessengerBundle/Registry/UcsMessageHandlerRegistry.php [hunk 1]
```diff
@@ -4,18 +4,20 @@
...
  */
-final class TraceableMessageHandlerRegistry
+final class UcsMessageHandlerRegistry
 {
-    /** @var array<string, TraceableMessageHandlerInterface> */
+    /** @var array<string, MessageHandlerInterface> */
     private array $handlers = [];
 
-    public function registerHandler(TraceableMessageHandlerInterface $handler, string $consumerName): void
+    public function registerHandler(MessageHandlerInterface $handler, string $consumerName): void
```
Comment (LaurentPetard) [inline src/Akeneo/Tool/Bundle/MessengerBundle/Registry/UcsMessageHandlerRegistry.php:20]: To see if we still want a specific interface, or no interface at all (as Symfony Messenger)

File: src/Akeneo/Tool/Bundle/MessengerBundle/Registry/UcsMessageHandlerRegistry.php [hunk 2]
```diff
@@ -4,18 +4,20 @@
...
+use Symfony\Component\Messenger\Handler\MessageHandlerInterface;
 
 /**
+ * @todo To rename
+ *
  * @copyright 2023 Akeneo SAS (https://www.akeneo.com)
  * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
  */
-final class TraceableMessageHandlerRegistry
+final class UcsMessageHandlerRegistry
```
Comment (LaurentPetard) [inline src/Akeneo/Tool/Bundle/MessengerBundle/Registry/UcsMessageHandlerRegistry.php:15]: To rename

File: src/Akeneo/Tool/Bundle/MessengerBundle/Registry/UcsMessageHandlerRegistry.php (hunk 1 above)
Comment (LaurentPetard) [reply src/Akeneo/Tool/Bundle/MessengerBundle/Registry/UcsMessageHandlerRegistry.php:20]: No interface

File: src/Akeneo/Tool/Bundle/MessengerBundle/Registry/ProcessMessageHandlerRegistry.php [hunk 3]
```diff
@@ -4,18 +4,16 @@
...
  */
-final class TraceableMessageHandlerRegistry
+final class ProcessMessageHandlerRegistry
 {
-    /** @var array<string, TraceableMessageHandlerInterface> */
+    /** @var object[] */
     private array $handlers = [];
 
-    public function registerHandler(TraceableMessageHandlerInterface $handler, string $consumerName): void
+    public function registerHandler(object $handler, string $consumerName): void
```
Comment (nmarniesse) [inline src/Akeneo/Tool/Bundle/MessengerBundle/Registry/ProcessMessageHandlerRegistry.php:16]: As we lost the type checking, can we check at least if `$handler` is callable in the body of the method? 
https://www.php.net/manual/en/function.is-callable.php

**Review by nmarniesse — APPROVED** [review]

**Comment by nmarniesse** [comment]
When we consume a message with the wrong bug (ex: we forgot to specify the bus), what happen?
The best would be to have an error to know something is wrong

**Comment by nmarniesse** [comment]
> When we consume a message with the wrong bug (ex: we forgot to specify the bus), what happen? The best would be to have an error to know something is wrong

We have a warning "No handler found" in dev environments, displayed even if we are not in verbose.

**Review by FabienD — APPROVED** [review]

**Pull request merged**